cmd/injector.go

```diff
@@ -49,11 +51,92 @@ func (e *FieldNotMatched) Is(err error) (matched bool) {
 	return
 }
 
+// TypeError used to report resource field cast error.
+type TypeError struct {
+	Field  *Field
+	Reason string
+	Object any
+}
+
+func (e *TypeError) Error() (s string) {
+	return fmt.Sprintf(
+		"Resource injector: cast failed. field=%s type=%s reason=%s, object:%v",
+		e.Field.Name,
+		e.Field.Type,
+		e.Reason,
+		e.Object)
+}
+
+func (e *TypeError) Is(err error) (matched bool) {
+	var inst *TypeError
+	matched = errors.As(err, &inst)
+	return
+}
+
 // Field injection specification.
 type Field struct {
-	Name string `json:"name"`
-	Path string `json:"path"`
-	Key  string `json:"key"`
+	Name    string `json:"name"`
+	Path    string `json:"path"`
+	Key     string `json:"key"`
+	Type    string `json:"type"`
+	Default any    `json:"default"`
+}
+
+// cast returns object cast as defined by Field.Type.
+func (f *Field) cast(object any) (cast any, err error) {
+	cast = object
+	if f.Type == "" {
+		return
+	}
+	defer func() {
+		if err != nil {
+			err = &TypeError{
+				Field:  f,
+				Reason: err.Error(),
+				Object: object,
+			}
+		}
+	}()
+	switch strings.ToLower(f.Type) {
+	case "string":
+		cast = fmt.Sprintf("%v", object)
+	case "integer":
```

Comment:
Consider supporting `case "integer" | "int"` and `case "boolean" | "bool"`

Reply:
Thinking longer form of integer|boolean|string would be more language (Go) agnostic.  Also, supporting the shorter forms would be more to document.  Seemed like less/simpler choices would be best.